src/main/java/by/bookify/image/service/GridFsMongoImageService.java

```diff
@@ -0,0 +1,66 @@
+package by.bookify.image.service;
+
+import by.bookify.image.exception.ImageNotFoundException;
+import by.bookify.image.exception.ImageUploadException;
+import com.mongodb.BasicDBObject;
+import com.mongodb.DBObject;
+import com.mongodb.client.gridfs.model.GridFSFile;
+import lombok.AllArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.core.io.InputStreamResource;
+import org.springframework.data.mongodb.core.query.Criteria;
+import org.springframework.data.mongodb.core.query.Query;
+import org.springframework.data.mongodb.gridfs.GridFsTemplate;
+import org.springframework.stereotype.Service;
+import org.springframework.web.multipart.MultipartFile;
+
+import java.io.IOException;
+import java.io.InputStream;
+
+@Service
+@AllArgsConstructor
+@Slf4j
+public class GridFsMongoImageService implements ImageService {
+
+    private GridFsTemplate gridFsTemplate;
+
+    @Override
+    public void uploadImage(Long bookId, MultipartFile file) {
+        DBObject metaData = createBasicDBObject(bookId);
+        gridFsTemplate.delete(new Query(Criteria.where("metadata.id").is(bookId)));
+        gridFsTemplate.store(fileToInputStream(file, bookId), file.getName(), file.getContentType(), metaData);
+    }
+
+    @Override
+    public InputStreamResource findImage(Long bookId) {
+        GridFSFile file = gridFsTemplate.findOne(new Query(Criteria.where("metadata.id").is(bookId)));
```

Comment:
Extract to variable. And also try to write Query annotation in controller

Reply:
In controller? May be you mean repository?

Reply:
I tried - it's not good idea, in my opinion

Reply:
Hahahah, it all was beer. I meant to write in "repository" not in a "controller"